clone/clone.go

```diff
@@ -53,8 +57,8 @@ in multiple places (Type IIG enzymes) such as BcgI.
 
 ******************************************************************************/
 
-// CloneSequence is a simple struct that can carry a circular or linear DNA sequence.
-type CloneSequence struct {
+// Part is a simple struct that can carry a circular or linear DNA sequence.
+type Part struct {
```

Comment:
```suggestion
type DNASequence struct {
```
Part is maybe too generic

Reply:
This was on @Koeng101's suggestion. It was originally called `CloneSequence` but would end up being written as `clone.CloneSequence` when used from another package. I couldn't have it be named `clone.Sequence` since it'd get confusing because the main poly package contains the `Sequence` struct for annotated sequences. I suggested `clone.Vector` but semantically `clone.Part` is most correct. I'll add some DNA part terminology to the comments to clarify this for future devs.

Reply:
I think Part is the most correct domain-wise, but I think jkhales is right that it is too generic for folks not building DNA. I would say keep Part but add a much larger comment.